test/server/cards/01_SOR/units/FettsFiresprayPursuingTheBounty.spec.ts

```diff
@@ -0,0 +1,176 @@
+            });
+        });
+
+        describe('Fett\'s Firespray\'s When Played ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['fetts-firespray#pursuing-the-bounty'],
+                        groundArena: ['jango-fett#renowned-bounty-hunter'],
+                        leader: 'sabine-wren#galvanized-revolutionary'
+                    },
+                    player2: {
+                        leader: 'chirrut-imwe#one-with-the-force'
+                    }
+                });
+            });
+
+            it('should be ready as we control Jango Fett', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.fettsFirespray);
+                expect(context.fettsFirespray.exhausted).toBeFalse();
+                expect(context.player2).toBeActivePlayer();
+            });
+        });
+
+        describe('Fett\'s Firespray\'s action ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['jango-fett#renowned-bounty-hunter', 'scout-bike-pursuer'],
+                        spaceArena: ['fetts-firespray#pursuing-the-bounty'],
+                        resources: 6
+                    },
+                    player2: {
+                        groundArena: ['battlefield-marine', 'echo-base-defender']
+                    }
+                });
+            });
+
+            it('should be exhaust a non-unique unit', function () {
+                const { context } = contextRef;
+
+                // exhaust battlefield marine
+                context.player1.clickCard(context.fettsFirespray);
+                expect(context.player1).toHaveEnabledPromptButtons(['Attack', 'Exhaust a non-unique unit']);
+                context.player1.clickPrompt('Exhaust a non-unique unit');
+                expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.scoutBikePursuer, context.echoBaseDefender]);
+                context.player1.clickCard(context.battlefieldMarine);
+
+                // fett's firespray should not be exhaust and 2 resources was spent
+                expect(context.fettsFirespray.exhausted).toBeFalse();
+                expect(context.player1.countExhaustedResources()).toBe(2);
+                context.player2.passAction();
+
+                // attack with fett's firespray
+                context.player1.clickCard(context.fettsFirespray);
+                context.player1.clickPrompt('Attack');
+                expect(context.fettsFirespray.exhausted).toBeTrue();
+                context.player2.passAction();
+
+                // we still can do his exhaust action even if he's exhausted
```

Comment:
These are all great! Can we do two additional tests since Firespray ability is one of the few that doesn't have the cost of exhausting the controller, just to test that the action ability cost system is working correctly.

1. Confirm that you can still pay the 2 for the ability even if all units are exhausted
2. If all units are exhausted and the controller does not have 2 resources to spend, ability is not available